packages/ui/src/expandable-box.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react'
+import { StateType, StatefulPluginEditorProps } from '@edtr-io/core'
+
+import { styled, faCaretSquareDown, faCaretSquareUp, Icon } from '.'
+
+export const expandableBoxState = StateType.object({
+  title: StateType.string(''),
+  content: StateType.child()
+})
+export class ExpandableBox extends React.Component<
+  ExpandableBoxProps,
+  HintState
+> {
+  public state: HintState = { hideContent: true }
+
+  public render() {
+    const { state, kind, editable } = this.props
+    const hideContent = !editable && this.state.hideContent
+
+    return (
+      <this.Wrapper hideContent={hideContent} kind={kind}>
+        <this.Toggle onClick={this.onToggle} kind={kind}>
+          {!editable ? (
+            hideContent ? (
+              <Icon icon={faCaretSquareDown} />
+            ) : (
+              <Icon icon={faCaretSquareUp} />
+            )
+          ) : null}
+          <a>
+            {kind} {this.showTitle()}
+          </a>
+        </this.Toggle>
+        <this.Content hideContent={hideContent}>
+          {state.content.render()}
+        </this.Content>
+      </this.Wrapper>
+    )
+  }
+
+  private showTitle(): React.ReactNode {
+    const { state, editable, kind } = this.props
+
+    if (editable) {
+      return (
+        <input
+          onChange={e => state.title.set(e.target.value)}
+          value={state.title()}
+          placeholder={
+            kind === 'Spoiler' ? 'Titel eingeben' : 'Zusätzlicher Name'
+          }
+        />
+      )
+    }
+    if (!state.title()) {
+      return null
+    }
+    return <React.Fragment>({state.title()})&nbsp;</React.Fragment>
+  }
+
+  private onToggle = () => {
+    this.setState({ hideContent: !this.state.hideContent })
+  }
+
+  private Wrapper = styled.div<{ hideContent?: boolean; kind?: string }>(
+    ({ hideContent, kind }) => {
+      return {
+        marginTop: '12px',
+        marginBottom: '20px',
+        border: hideContent
+          ? '1px solid transparent'
+          : kind === 'Lösung'
```

Comment:
we should handle that differently, e.g. by passing the colors to `ExpandableBox`. `ExpandleBox` shouldn't need to know the different possible kinds, especially, when we expose that via `@edtr/ui`.

Alternatively: allow theming

Reply:
@Knorrke We should discuss how we want to handle Theming of such shared components.

Reply:
We shortly discussed this:
- our `@edtr-io/ui` components expose a theme
- plugins expose a theme
- Editor theme: `{ ui: { expandableBox: {}, ... }, plugins: { text: { ... } }`

Reply:
Going to do this in a different PR first before merging #55